client/src/main/java/io/split/engine/sse/dtos/CommonChangeNotification.java

```diff
@@ -16,25 +15,29 @@
 import static io.split.engine.sse.utils.DecompressionUtil.gZipDecompress;
 import static io.split.engine.sse.utils.DecompressionUtil.zLibDecompress;
 
-public class FeatureFlagChangeNotification extends IncomingNotification {
+public class CommonChangeNotification<Y> extends IncomingNotification {
     private static final Logger _log = LoggerFactory.getLogger(SegmentSynchronizationTaskImp.class);
     private final long changeNumber;
     private long previousChangeNumber;
-    private Split featureFlagDefinition;
     private CompressType compressType;
+    private Y definition;
+    private Class _definitionClass;
 
-    public FeatureFlagChangeNotification(GenericNotificationData genericNotificationData) {
-        super(Type.SPLIT_UPDATE, genericNotificationData.getChannel());
+    public CommonChangeNotification(GenericNotificationData genericNotificationData,
+                                    Class definitionClass) {
```

Comment:
i think we don't need this.
```suggestion
    public CommonChangeNotification(GenericNotificationData genericNotificationData) {
```